pkg/proxy/healthcheck/service_health.go

```diff
@@ -183,7 +183,7 @@ func (hcI *hcInstance) listenAndServeAll(hcs *server) error {
 		go func(hcI *hcInstance, listener net.Listener, httpSrv httpServer) {
 			// Serve() will exit when the listener is closed.
 			klog.V(3).InfoS("Starting goroutine for healthcheck", "service", hcI.nsn, "address", listener.Addr())
-			if err := httpSrv.Serve(listener); err != nil {
+			if err := httpSrv.Serve(listener); err != nil && !strings.Contains(err.Error(), "use of closed network connection") {
```

Comment:
It's not good to check for specific error text. Is there some other way to do this?
(IIRC in one of these situations you can call two separate methods instead of one, and the first one can return a (valid) error but the second one doesn't.)

Reply:
yeah, in another angle, is this even an error log message or an informative one?

Reply:
> It's not good to check for specific error text. Is there some other way to do this?

It could return `http.ErrServerClosed` but we need to change to close the http server, instead of the listener.

> (IIRC in one of these situations you can call two separate methods instead of one, and the first one can return a (valid) error but the second one doesn't.)

I don't find the two separate methods, did you mean `ListenAndServe` vs. `net.Listen() + Serve()`? If yes, the original code already did so.

> yeah, in another angle, is this even an error log message or an informative one?

I guess this is meant for error message as there is an informative one in next line: https://github.com/kubernetes/kubernetes/blob/7eb9a75bf914428e7579d2497722f71be3ef1812/pkg/proxy/healthcheck/service_health.go#L190

I have updated the patch to close the server and check error in a more graceful way, PTAL.